### PR TITLE
chore: upgrade to pnpm 8, fix list types in prisma schema

### PR DESCRIPTION
**Pull request by fiam**
- Update engines.pnpm in package.json
- Update pnpm version in node action
- Remove engine-strict=true from .npmrc
- Fix handling of list types in arguments to prisma operations

#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by ggeijteman-wg — APPROVED** [review]

File: .npmrc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 strict-peer-dependencies=false # If this is enabled, commands will fail if there is a missing or invalid peer dependency in the tree.
 fetch-retries="10"
-engine-strict=true
-# true is the default for PNPM 8, but we're still on 7.
-# Enabling this breaks our deployment flow.
+# Enabling this breaks our deployment because it generates peer
```
Comment (Aenimus) [inline .npmrc:3]: "This" has no context after you remove it

Comment (fiam) [reply .npmrc:3]: It refers to the next line where we disable `dedupe-peer-dependents`. Maybe there's a better way to phrase it?

Comment (Aenimus) [reply .npmrc:3]: "Enabling dedupe-peer-dependents..."

**Review by Aenimus — APPROVED** [review]

File: .npmrc (hunk 1 above)
Comment (fiam) [reply .npmrc]: Thanks, that certainly reads better! I've rewritten it as you suggested.

**Comment by Aenimus** [comment]
@fiam CI failures

**Comment by fiam** [comment]
> @fiam CI failures

Yeah, I’m on them. Thanks for the reminder.

**Pull request merged**